src/elements/name_section.rs

```diff
@@ -263,7 +263,17 @@ impl LocalNameSubsection {
 
 		let max_locals = module
 			.code_section()
-			.map(|cs| cs.bodies().iter().map(|f| f.locals().len()).max().unwrap_or(0))
+			.map(|cs|
+                cs.bodies()
+                    .iter()
+                    .map(|f|
+                        f.locals()
+                            .iter()
+                            .map(|l| l.count() as usize)
+                           .max()
+                          .unwrap_or(0))
+                    .max()
+                    .unwrap_or(0))
```

Comment:
First the whitespace issue: The indentation is off. I guess some funny tab vs spaces issue. Also please close a parentheses on its own line if its opened on its own. This makes is hard to read.

I have doubts about the correctness of this code. This **should** determine the maximum amount of locals over all bodies, right? As far as I understand the new code it will only determine the maximum amount of compressed locals for every function which is wrong.

Reply:
Maybe I am missing something, but is an “uncompressed local” not just one with `l.count() == 1`?

Reply:
You are not using `cargo-fmt`, right? If I apply this, I get lots of changes in all files?

Reply:
Tabs fixed, sorry.

> also please close a parentheses on its own line if its opened on its own.

I  am merely sticking to the coding convention around (see `max_signature_args` above)…

Reply:
> Maybe I am missing something, but is an “uncompressed local” not just one with `l.count() == 1`?

That is correct. If I am not misinterpreting your code it just returns the the biggest `count()` over all functions. While it should return the maximum number of locals over all functions. In other words: In a program that has only "uncompressed locals" it will return 1.

Reply:
Oh, right, now I see it. My bad.